### ISSUE TITLE
devdocs now requires emacs 27.1

### ISSUE THREAD

**Issue by mbertheau** [syl20bnr/spacemacs#14820]
Up until a few days ago devdocs was this package: https://github.com/xuchunyang/DevDocs.el/blob/master/devdocs.el

Now it is this package: https://elpa.gnu.org/packages/devdocs.html https://github.com/astoff/devdocs.el

The new package requires emacs 27.1: https://github.com/astoff/devdocs.el/blob/8a3f4e00e0a2113c604c631f81f7db90c1b56d5b/devdocs.el#L8

devdocs is in spacemacs-misc: https://github.com/syl20bnr/spacemacs/blob/1ce4a7fd66a4f19c8b9008798d526a754b213f94/layers/%2Bspacemacs/spacemacs-misc/packages.el#L52

How do we handle this?

**Comment by lebensterben** [comment]
make the last compatible one a local package

**Comment by mbertheau** [comment]
How do we handle this for all spacemacs users that use emacs 26?

**Comment by lebensterben** [comment]
I said local package.
we've plenty of local packages already

**Comment by lebensterben** [comment]
example of local package: https://github.com/syl20bnr/spacemacs/blob/af86a142ebeff3bfa0c5cb41d47455af3cd68eab/layers/%2Bspacemacs/spacemacs-defaults/packages.el#L41

For Emacs < 27, use a local package. Otherwise use the upstream one.

**Comment by emacs18** [comment]
This was discussed recently on emacs-dev list.  The new package has far more features than the old one.
Also https://github.com/blahgeek/emacs-devdocs-browser is similar package which I test drove.
I assume that emacs-devdocs-browser and devdocs packages provide similar features.

I think these new document browsing packages may be more convenient for some use cases than https://github.com/dash-docs-el/dash-docs which I've used for a while.

So I hope that we can find a way to provide these really cool features to those spacemacs users willing to use newer emacs versions.

**Comment by thanhvg** [comment]
I think we should just spin off devdocs to an optional layer for Emacs 27. The old devdocs package was very simple, all it did was to open devdocs.io with current symbol. I don't think anyone would miss it.  While local package should work, I don't like the idea that Spacemacs will take the maintenance job of a deprecate one. I also prefer `devdoc-browser` (https://github.com/blahgeek/emacs-devdocs-browser) over the new `devdocs`.

But this trouble could be avoided if we only supported the latest stable Emacs in the first place.

**Comment by emacs18** [comment]
After few minutes of testing driving new `devdocs` and `emacs-devdocs-browser`, I also prefer the latter.
Both provide completions fine, e.g., typing "os open" with python documentation provided "os.open()" link.
`emacs-devdocs-browser` found the exact line where the documentation for `os.open()`began.
However `devdocs` brought up the beginning of "os" module documentation so that I had to scroll down to location documentation for "os.open()".

Unfortunately `emacs-devdocs-browser` is not yet in any package archive, so I installed it using `straight.el` via this recipe:

```elisp
(straight-use-package '(devdocs-browser :type git :host github :repo "blahgeek/emacs-devdocs-browser"))
```
I'm sure that one can also use `package.el` to install this as easily.

**Comment by lebensterben** [comment]
yes let's migrate to `devdocs-browser`. I just checked it and it has good code quality.

**Comment by lebensterben** [comment]
We're very likely to elevate minimum emacs version to 27.1 this year.
When we bumped it up to 26.1, the reason was that it's the version offered on Debian stable branch (buster).

But then it's bumped to 26.3, which breaks the soft target of matching Debian's emacs version.

Debian bullseye is in RC1 now: https://lists.debian.org/debian-devel-announce/2021/04/msg00011.html
When it's released we'll have no reason to stay with emacs < 27

**Comment by mbertheau** [comment]
Sounds good. Until that happens we can use a local package. Here's a PR to that effect: https://github.com/syl20bnr/spacemacs/pull/14823

**Comment by lebensterben** [comment]
Fixed in #14823
